events/permissions.py

```diff
@@ -93,6 +93,11 @@ def has_permission(self, request, view):
             return False
 
 
+class GuestRetrieve(BasePermission):
+    def has_permission(self, request, view):
+        return request.method == "GET" and view.kwargs.get("pk")
```

Comment:
Afaik this can also be done with:
```suggestion
        return view.action == "retrieve"
```

Reply:
Ah, makes sense. Changed and tested that is works as expected.